Update to 5.12.4

jschueller opened the pull request:
<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [ ] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

mingwandroid commented on recipe/meta.yaml:
```diff
@@ -3,21 +3,19 @@
-  - url: http://download.qt.io/official_releases/qt/5.12/{{ version }}/single/qt-everywhere-src-{{ version }}.zip  # [win]
-    md5: 4649d4e51ca836fbde08565582353140  # [win]
+  - url: http://download.qt.io/official_releases/qt/5.12/{{ version }}/single/qt-everywhere-src-{{ version }}.tar.xz
+    sha256: 85da5e0ee498759990180d5b8192efaa6060a313c5018b772f57d446bdd425e1
```
Where did you get this from? The Qt project publishes only md5s:

http://download.qt.io/official_releases/qt/5.12/5.12.4/md5sums.txt

mingwandroid commented:
I'm not sure if you want to hold off as I am actively working on Qt over at https://github.com/mingwandroid/qt-feedstock

I started from this PR, but I've done a lot of cleanup of the qtbase patches with more changes to come.

If you want I can make PRs to this PR, but if we diverge then that'll become trickier.

jschueller commented:
hi @mingwandroid,

I think I'm stuck anyways, I've got this error on osx:
```
 2019-08-22T13:10:10.3158870Z ssl/qsslsocket_mac.cpp:509:10: error: use of undeclared identifier 'kTLSProtocol13'
2019-08-22T13:10:10.3159580Z     case kTLSProtocol13:
```

mingwandroid commented:
That's exactly where I'm up to on macOS at present too. I am happy to peek at this.

I have a linux-64 build (which uses defaults then conda-forge as build deps, but I'm hoping that won't matter these days) on anaconda.org/rdonnelly now. It failed a test but I ignored that for now and uploaded it from my broken folder anyway!

jschueller commented:
its due to this patch:
https://github.com/qt/qtbase/commit/a389a9f4404c5dc400d23d4fe2a70557d161731a

mingwandroid commented:
For AD we have a conda_build_config.yaml file local to the recipe containing:

```
CONDA_BUILD_SYSROOT:
  - /opt/MacOSX10.12.sdk        # [osx]
```

Pinging @conda-foge/core, are we OK with (or have we already?) setting MacOS 10.12 as the minimum supported macOS version for using any Qt based things?

mingwandroid commented:
Trying something whacky, sure to fail:

```
CONDA_BUILD_SYSROOT:       # [osx]
  - /opt/MacOSX10.12.sdk   # [osx]
MACOSX_DEPLOYMENT_TARGET:  # [osx]
  - '10.10'                # [osx]
```

isuruf commented:
At conda-forge, we only set `MACOSX_DEPLOYMENT_TARGET`.

```
MACOSX_DEPLOYMENT_TARGET:  # [osx]
  - '10.12'                # [osx]
```

CI script will read this variable, download the sdk somewhere, extract it and set `CONDA_BUILD_SYSROOT` automatically. So `CONDA_BUILD_SYSROOT` should not be added to the conda-forge recipe. (Only `MACOSX_DEPLOYMENT_TARGET`)

jschueller commented:
it's already set in build_config.yaml, and its also 10.12...

mingwandroid commented:
I'm making some modifications to build.sh over at my master branch so that it parses CONDA_BUILD_SYSROOT to determine the SDK version which is then used in the rest of build.sh so that it all comes from that source (which conda-forge stuff is feeding into..)

Does that sound OK?

isuruf commented:
If you can check that the basename of `CONDA_BUILD_SYSROOT` is `MacOSX10.XX.sdk` and if not overwrite it with `/opt/MacOSX10.XX.sdk`, then that should work for conda-forge.

mingwandroid commented:
Currently (WIP!) https://github.com/mingwandroid/qt-feedstock/commit/2ecb450e0ea731fe5d5a2f22a46321efbc64e086

mingwandroid commented:
Less utterly broken now: https://github.com/mingwandroid/qt-feedstock/commit/164c92d15c71a87462e42a9ff9df3942c3145ef8

jschueller commented:
oh it's actually using 10.13/10.14:
```
macOS version .......................... 10.13.6
2019-08-22T20:23:14.0074010Z   Xcode version .......................... 8.3.3
2019-08-22T20:23:14.0074350Z   Clang version .......................... clang version 4.0.1 (tags/RELEASE_401/final)
2019-08-22T20:23:14.0074700Z   macOS SDK version ...................... 10.14.1
2019-08-22T20:23:14.0075040Z   macOS minimum deployment target ........ 10.12
```

mingwandroid commented:
That's what the xcrun etc patch is *meant* to address.

mingwandroid commented:
Ah, well, we need to either revert the support for TLS1.3 since it's only in macOS 10.13 and above of make 10.13 the minimum for Qt. What is the general opinion here? For AD, I think we'd still like to stick to 10.12 and not provide TLS1.3 support by adding a reversion patch.

mingwandroid commented:
Useful: http://codeworkshop.net/objc-diff/sdkdiffs/macos/10.13/IOSurface.html

mingwandroid commented:
I'm making good progress over on my master branch by the way. Also tidying things up a good bit. If may be time to think about trying to upstream the less controversial bits too.

jschueller commented:
so do we need your CONDA_BUILD_SYSROOT changes here, or you need it only on your side ?
it seems the build timeouts when building the huge js runtime (chromium ?!), cannot it be built separately ? who actually provides the binaries then ?

mingwandroid commented:
More patches from me:

https://github.com/mingwandroid/qt-feedstock/commit/50afc643c10a15e589942f40a14afb5810944372
https://github.com/mingwandroid/qt-feedstock/commit/4e2f5f371dd275a6eb6fa0c1b049aee765cd811f
https://github.com/mingwandroid/qt-feedstock/commit/fd230d9da5584fde435a1d27f0616fd1fa5fdc34

.. but swapping partches like this is a 1980s way to do things and I would prefer in general to either make PR to your fork or for you to join in at my qt-feedstock repo (I can give you push access if you want)..

jschueller commented:
Or you could send your own PR since you did the heavy lifting here, I'll close mine.
Note that I modified my first commit to keep the md5, and that I removed my commit that skips the xcrun patch (you also updated it in your branch).

jschueller commented:
With your latest kiosurface patch the build goes fine on OSX!
Linux and Windows builds timeout in chromium.
What do we do now ?

mingwandroid commented:
Timing out is unfortunate, I don't know what we do in this case, pinging @mariusvniekerk and @ocefpaf about that .. (note I don't think Windows is ready yet!)

ocefpaf commented:
> Timing out is unfortunate, I don't know what we do in this case, pinging @mariusvniekerk and @ocefpaf about that .. (note I don't think Windows is ready yet!)

We build it locally :smile: 

I'm in a workshop until September 1st but I can do both the Windows and Linux builds after that.

jschueller commented:
@ocefpaf that would be awesome

jschueller commented:
@ocefpaf I successfully run the Linux build, could you test on windows ?

ocefpaf commented:
> @ocefpaf I successfully run the Linux build (with the build-locally.py script),

Awesome!

> could you test on windows ?

Testing it later today. Stay tuned!

jschueller commented:
still compiling @ocefpaf ? :)

ocefpaf commented:
> still compiling @ocefpaf ? :)

Nope 😄 
Had major Windows issues these day. Compiling now! Will report in a few hours. Stay tuned.

ocefpaf commented:
@jschueller the Windows build failed after a few hours with:

```
%SRC_DIR%>SET "CMAKE_GENERATOR=Visual Studio 14 2015 Win64"

%SRC_DIR%>CALL "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\..\..\VC\vcvarsall.bat" amd64
Info: creating stash file %SRC_DIR%\test\.qmake.stash

Microsoft (R) Program Maintenance Utility Version 14.00.24210.0
Copyright (C) Microsoft Corporation.  All rights reserved.

        "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64\nmake.exe" -f Makefile.Release

Microsoft (R) Program Maintenance Utility Version 14.00.24210.0
Copyright (C) Microsoft Corporation.  All rights reserved.

        cl -c -nologo -Zc:wchar_t -FS -Zc:rvalueCast -Zc:inline -Zc:strictStrings -Zc:throwingNew -O2 -MD -W3 -w34100 -w34189 -w44996 -w44456 -w44457 -w44458 -wd4577 -wd4467 -EHsc -DUNICODE -D_UNICODE -DWIN32 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -DWIN64 -DQT_NO_DEBUG -DQT_CHARTS_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_CORE_LIB -DNDEBUG -I. -I..\..\_test_env\Library\include\qt -I..\..\_test_env\Library\include\qt\QtCharts -I..\..\_test_env\Library\include\qt\QtWidgets -I..\..\_test_env\Library\include\qt\QtGui -I..\..\_test_env\Library\include\qt\QtANGLE -I..\..\_test_env\Library\include\qt\QtCore -Irelease -I..\..\_test_env\Library\mkspecs\win32-msvc -Forelease\ @C:\Users\filipe\AppData\Local\Temp\nm3378.tmp
main.cpp
        link /NOLOGO /DYNAMICBASE /NXCOMPAT /INCREMENTAL:NO /SUBSYSTEM:CONSOLE "/MANIFESTDEPENDENCY:type='win32' name='Microsoft.Windows.Common-Controls' version='6.0.0.0' publicKeyToken='6595b64144ccf1df' language='*' processorArchitecture='*'" /MANIFEST:embed /OUT:release\hello.exe @C:\Users\filipe\AppData\Local\Temp\nm3752.tmp

Microsoft (R) Program Maintenance Utility Version 14.00.24210.0
Copyright (C) Microsoft Corporation.  All rights reserved.

        "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64\nmake.exe" -f Makefile.Release clean

Microsoft (R) Program Maintenance Utility Version 14.00.24210.0
Copyright (C) Microsoft Corporation.  All rights reserved.

        del release\moc_predefs.h
Could Not Find C:\Users\filipe\Miniconda3\conda-bld\qt_1567700225587\test_tmp\test\release\moc_predefs.h
        del release\main.obj
        "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64\nmake.exe" -f Makefile.Debug clean

Microsoft (R) Program Maintenance Utility Version 14.00.24210.0
Copyright (C) Microsoft Corporation.  All rights reserved.

        del debug\moc_predefs.h
Could Not Find C:\Users\filipe\Miniconda3\conda-bld\qt_1567700225587\test_tmp\test\debug\moc_predefs.h
        del debug\main.obj
Could Not Find C:\Users\filipe\Miniconda3\conda-bld\qt_1567700225587\test_tmp\test\debug\main.obj
        del debug\hello.vc.pdb debug\hello.ilk debug\hello.idb
Could Not Find C:\Users\filipe\Miniconda3\conda-bld\qt_1567700225587\test_tmp\test\debug\hello.vc.pdb
Project ERROR: Unknown module(s) in QT: webengine
Tests failed for qt-5.12.4-h506e8af_0.tar.bz2 - moving package to C:\Users\filipe\Miniconda3\conda-bld\broken
```

jschueller commented:
Maybe qtwebengine did not succeed, do you still got the configuration log to look for an error ?
at least the configuration part should should look like this, like on azure:
```
2019-08-30T17:14:39.7314283Z Qt WebEngine:
2019-08-30T17:14:39.7314579Z   Embedded build ......................... no
2019-08-30T17:14:39.7314858Z   Pepper Plugins ......................... yes
2019-08-30T17:14:39.7315159Z   Printing and PDF ....................... yes
2019-08-30T17:14:39.7315482Z   Proprietary Codecs ..................... no
2019-08-30T17:14:39.7315773Z   Spellchecker ........................... yes
2019-08-30T17:14:39.7316073Z   Native Spellchecker .................... no
2019-08-30T17:14:39.7316345Z   WebRTC ................................. yes
2019-08-30T17:14:39.7316643Z   Use System Ninja ....................... yes
2019-08-30T17:14:39.7316918Z   Geolocation ............................ yes
2019-08-30T17:14:39.7317327Z   WebChannel support ..................... yes
2019-08-30T17:14:39.7317611Z   Use v8 snapshot ........................ yes
2019-08-30T17:14:39.7317905Z   Kerberos Authentication ................ yes
```

ocefpaf commented:
> Maybe qtwebengine did not succeed, do you still got the logs to look for an error ?
> at least the configuration part should should look like this, like on azure:

Here are the logs:

[log.txt](https://github.com/conda-forge/qt-feedstock/files/3580710/log.txt)
[err.txt](https://github.com/conda-forge/qt-feedstock/files/3580712/err.txt)

I'll inspect them soon. Got to move to another task now.

jschueller commented:
see err.txt:
```
Qt WebEngine on Windows requires a Windows SDK version 10.0.16299 or newer.
QtWebEngine will not be built.
```

jschueller commented:
it seems the incremented it: https://code.qt.io/cgit/qt/qtwebengine.git/commit/?id=2025e8fa0dd60cc012c9b2a0135164c3535acc5d ???

ocefpaf commented:
> it fails when detecting the vs version (vs2017 required I think), good (azure):

I'll have some time with a Windows machine tomorrow and I'll install vs2017. Which PR should try? This one or #120?

jschueller commented:
this one should be ok

jschueller commented:
but, shouldnt you get vs2017 through conda, as listed in the meta.yaml ?

jschueller commented:
see #122
